### PR TITLE
fix(rules): Surface errors from Sentry Apps + reformat slack errors

### PR DESCRIPTION
**Pull request by leeandher**
See [API-2619](https://getsentry.atlassian.net/browse/API-2619)

This PR cleans up the errors that appear on the rules page for Sentry Apps and Slack.

I've also added a TODO to refactor the function that checks for Alert Rule Actions into a better wrapper, along with renaming one of the constants.

**New Slack Errors**:

Screen Shot 2022-04-12 at 10 55 26 AM

Screen Shot 2022-04-12 at 10 55 17 AM
 

**New Sentry App Errors**: 

Screen Shot 2022-04-12 at 3 58 04 PM

Screen Shot 2022-04-12 at 4 01 12 PM

### CODE REVIEW

File: src/sentry/constants.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def get_all_languages() -> List[str]:
     ]
 )
 
-SCHEMA_FORM_ACTIONS = frozenset(
+SENTRY_APP_ACTIONS = frozenset(
```
Comment (leeandher) [inline src/sentry/constants.py:259]: This name SCHEMA_FORM_ACTIONS isn't descriptive, nor helpful to identify what the action really is.

File: src/sentry/integrations/slack/notify_action.py [hunk 2]
```diff
@@ -95,19 +96,16 @@ def clean(self) -> dict[str, Any] | None:
             except ValidationError as e:
                 params = {"channel": self.data.get("channel"), "channel_id": channel_id}
                 raise forms.ValidationError(
-                    _(
-                        str(e),
-                    ),
+                    # ValidationErrors contain a list of error messages, not just one.
+                    self._format_error_message("; ".join(e.messages)),
```
Comment (leeandher) [inline src/sentry/integrations/slack/notify_action.py:100]: If there is only one error, the semi-colon is not shown.

File: src/sentry/mediators/external_requests/alert_rule_action_requester.py [hunk 3]
```diff
@@ -57,16 +73,11 @@ def _make_request(self) -> Mapping[str, Union[bool, str]]:
...
-            # NOTE: bool(e.response) is always False because non-2xx responses are Falsey.
-            text = e.response.text if e.response is not None else None
-            message = f"{self.sentry_app.name}: {text or DEFAULT_ERROR_MESSAGE}"
+            message = f"{self.sentry_app.name} Error: {self._get_error_message(e.response)}"
             # Bubble up error message from Sentry App to the UI for the user.
-            return {"success": False, "message": message}
-
-        body_raw = safe_urlread(request)
-        body = body_raw.decode() if body_raw else None
-        message = f"{self.sentry_app.name}: {body or DEFAULT_SUCCESS_MESSAGE}"
```
Comment (leeandher) [inline src/sentry/mediators/external_requests/alert_rule_action_requester.py:68]: @NisanthanNanthakumar  I don't believe we were using the message anywhere for successful alert rules created, so I removed any code that looked at the response, and just opted to always use the default success message.  Am I understanding it correctly?

**Review by ceorourke — APPROVED** [review]
LGTM, but I see you called out @NisanthanNanthakumar to look at a specific thing so maybe bug him to take a peek first :)

File: src/sentry/mediators/external_requests/alert_rule_action_requester.py (hunk 3 above)
Comment (leeandher) [reply src/sentry/mediators/external_requests/alert_rule_action_requester.py:68]: bumping this question @NisanthanNanthakumar

Comment (NisanthanNanthakumar) [reply src/sentry/mediators/external_requests/alert_rule_action_requester.py:68]: @leeandher yea, we don't currently show the success response text from the SentryApp in the UI. However, that may change in the future. I think it's best to surface the success response text in the API response. IMO, it doesn't really matter if the UI currently considers it to be unnecessary.

File: src/sentry/integrations/slack/notify_action.py [hunk 4]
```diff
@@ -57,6 +57,9 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
         # in the task (integrations/slack/tasks.py) if the channel_id is found.
         self._pending_save = False
 
+    def _format_error_message(self, message: str) -> Any:
```
Comment (NisanthanNanthakumar) [inline src/sentry/integrations/slack/notify_action.py:60]: I'd call this `_format_slack_error_message`

**Review by NisanthanNanthakumar — CHANGES_REQUESTED** [review]

File: src/sentry/mediators/external_requests/alert_rule_action_requester.py (hunk 3 above)
Comment (leeandher) [reply src/sentry/mediators/external_requests/alert_rule_action_requester.py:68]: Sure, added and wrote some tests for it as well.

**Comment by leeandher** [comment]
@NisanthanNanthakumar Do you mind re-reviewing this?

**Review by NisanthanNanthakumar — APPROVED** [review]
Looks good!!

**Pull request merged**